### PR TITLE
Refactor output_interval method to fix rubocop issues

### LINKED ISSUE #253: stats/perf-stat: resolve rubocop Metrics issues of output_interval method
### Reproduce
```
$ rubocop --only Metrics/AbcSize,Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity stats/perf-stat

stats/perf-stat:81:1: C: Metrics/AbcSize: Assignment Branch Condition size for output_interval is too high. [<16, 19, 18> 30.68/30]
def output_interval(prev_time, time) ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
stats/perf-stat:81:1: C: Metrics/CyclomaticComplexity: Cyclomatic complexity for output_interval is too high. [14/7]
def output_interval(prev_time, time) ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
stats/perf-stat:81:1: C: Metrics/PerceivedComplexity: Perceived complexity for output_interval is too high. [15/8]
def output_interval(prev_time, time) ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
### Reference
https://github.com/intel/lkp-tests/wiki/Coding-Standard#rubocop

### Acceptance Test
```
$ rubocop --only Metrics/AbcSize,Metrics/CyclomaticComplexity,Metrics/MethodLength,Metrics/PerceivedComplexity stats/perf-stat # has no issue of output_interval related code
$ rake spec spec=stats # pass
```

### PR DESCRIPTION
**Pull request by fongjiantan**
Fixes #253

### CODE REVIEW

File: stats/perf-stat [hunk 1]
```diff
@@ -81,21 +81,40 @@ end
...
                   interval, 1, 'i.')
+end
 
-  unless $interval_metrics_val.empty?
-    $interval_metrics_val.each do |key, value|
-      puts "i.#{key}: #{value}"
-    end
-  end
-
+def ignore_result(prev_time, time)
```
Comment (rli9) [inline stats/perf-stat:115]: thanks, perfer to use ignore_result? method

The code then is like
```
if time <= $stats_begin_time
  true
elsif prev_time <= $stats_begin_time
  true
else
  false
end
```

```

**Comment by fongjiantan** [comment]
@rli9  I have updated the code based on your review. Thanks

File: stats/perf-stat [hunk 2]
```diff
@@ -121,6 +102,31 @@ def output_interval(prev_time, time)
...
+  puts "time: #{time}"
+  $interval_results_sum.each do |key, value|
+    puts "i.#{key}: #{value / interval}"
+  end
+  calc_addon_keys($interval_results_sum, $interval_results_nr,
+                  interval, 1, 'i.')
+end
+
+def ignore_result?(prev_time, time)
+  if time <= $stats_begin_time
```
Comment (rli9) [inline stats/perf-stat:116]: thanks, based on this code, prefer here to have an early return
```
return true if time <= $stats_begin_time
```

Comment (fongjiantan) [reply stats/perf-stat]: Fixed, thanks!

**Pull request merged**